### ISSUE TITLE
Ordering API returning 404 to admin orders page

### ISSUE THREAD

**Issue by DavidAJohn** [DavidAJohn/FotoStorioMicroservices#22]
While updating the Admin.BlazorServer UI project, I noticed the admin page which shows customer orders provoked a 404 from the Ordering API (via the admin gateway) because there are no orders in the database (or none in the last x days).

I'm not sure this is the correct response when there are no orders to return, but there can be [huge debates](https://stackoverflow.com/questions/11746894/what-is-the-proper-rest-response-code-for-a-valid-request-but-empty-data) over this.

Ultimately, what does the CustomerOrders.razor page need to do? Is there value in returning a different response when there are no orders to return, as opposed to when there is a problem retrieving the orders?

**Comment by DavidAJohn** [comment]
Although this is a bit of an edge case - because it would probably only happen on a fresh database with no orders in it - I do think that returning a 200 OK response with an empty dataset is preferable in this case.

There are currently two client app pages that display order information: the **Store,BlazorWasm** shows a customer their previous orders (Pages/Account/Orders.razor) and in **Admin.BlazorServer** there is a page that shows admins a list of all orders (Pages/Orders/CustomerOrders.razor), which can be filtered via a dropdown. 

Both of those pages are actually already set up to deal with empty datasets being returned. They initially create an order list object as null, then there is branching logic to display a list of orders if the object becomes non null and the count of orders is > 0. If it is not null and the order count is 0, then it displays 'no orders were found'. If the order service (which makes the HTTP request to the API) throws an exception, the error message variable is filled and that is displayed on screen instead.

Looking at the Orders Controller in the Ordering API, it feels as if there was some rather careless copying and pasting of methods, but if the two methods that return lists of orders are changed to return a 200 OK with an empty list instead of a 404, then I think everything else will probably continue to work as intended.

**Comment by DavidAJohn** [comment]
While I was working on this, I noticed another little quirk in the Ordering API.

There are two methods in the order repository that return lists of orders to the controller endpoint methods that I mentioned above. The repository methods use LINQ to return orders via the db context, but they both end with a ToListAsync(), which always returns a list even if no items are found.

The null checks after this were obviously never being hit. It didn't matter because AutoMapper was handling the data returned to the controller methods either way, but I wanted it to be clear to the controller if there were no orders found so that it could intentionally return an empty list of orders (and perhaps log this fact for informational purposes).

As a result, I changed the null check for a check on the count of items in the list of orders, returning null if it equals zero.
